Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Cursors/BaseCursor.cs

```diff
@@ -141,7 +141,13 @@ public virtual void OnSourceDetected(SourceStateEventData eventData)
                     if (eventData.InputSource.Pointers[i].PointerId == Pointer.PointerId)
                     {
                         visibleSourcesCount++;
-                        SetVisibility(true);
+
+                        if (visibleSourcesCount == 1 &&
+                            InputSystem.GazeProvider.GazePointer.PointerId != Pointer.PointerId)
+                        {
+                            SetVisibility(isVisible);
```

Comment:
because sometimes it's not

Reply:
But `isVisible` should be the current visibility of the cursor. If it's false, this will set it to false again?

Reply:
no, because the [underlying logic checks if the value is the same before setting](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/2750/files#diff-0df32f8308189929c96b3a5e6b59ca08R120).

Reply:
Which it will always be after the first time, right?

[`SetVisibility` sets `isVisible` to be itself](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/2750/files#diff-0df32f8308189929c96b3a5e6b59ca08R110) (because it's the parameter). Then, this script uses the exact same value from then on.

Reply:
Try setting it to true and see if it makes a difference on the HoloLens.

That's mainly why I changed it to be `isVisible` instead of true in the gestures PR

Reply:
Doesn't the gaze pointer check right above filter out HoloLens anyway?

This being anything other than true doesn't make sense and adds a circular reference if the cursor sticks around after a source lost. Visibility will stay false from that point on.

Reply:
Yeah that makes sense. Would you like me to go ahead and update it?

Reply:
Seems like this bug was introduced in https://github.com/Microsoft/MixedRealityToolkit-Unity/commit/d0a80f02127e2dfcedcaf97710aab129ff1898a9#diff-0df32f8308189929c96b3a5e6b59ca08R144 by making visibility tied to source detected/lost. Why was that added?

Reply:
So that motion controller cursors would be enabled correctly.

Reply:
How were they not enabling correctly? I remember them working fine before that change.

Reply:
Thanks!